app/controllers/lists_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ListsController < ApplicationController
+
+  # users must be signed in before any lists_controller method
+  before_action :authenticate_user!
+
+  def index
+    @lists = List.all
+  end
+
+  def show
+    @list = List.find(params[:id])
```

Comment:
There's a potential security vulnerability here and in some of the other actions that an attacker could take advantage of through tampering with the `id` sent through the requests (e.g. by modifying the URL). The problem lies in fetching by id without limiting to just the `current_user`'s lists:

```ruby
List.find(params[:id]) # In console use List.find(some_id_value)
# compare to
current_user.lists.find(params[:id]) # In console use User.last.lists.find(some_id_value)
```

Consider experimenting with those two queries in the rails console to see the generated SQL and how one of the queries doesn't protect the user's data.

This would be a good opportunity to introduce a controller spec that tests for these security issues. Perhaps we can tackle that in the next call.